Marketplace/controllers/api/apiUserController.js

```diff
@@ -1,11 +1,49 @@
-const db = require("../../database/models");
-const entidad = db.Usuario;
+const usuarioRepository = require("../../repositories/usuarioRepository");
 
 module.exports = {
-	listado: (req, res) => {
-
-	},
-	detalle: (req, res) => {
-		
-	},
+	listado: async (req, res) => {
+        let usuarios = await usuarioRepository.ObtenerTodos();
+        let usuarioRespuesta = usuarios.map(x => {
+            return {
+                id: x.id,
+                nombre: x.nombre,
+				email: x.email
+            }
+        });
+        let respuesta = {
+            meta: {
+				   TotalUsuarios: usuarios.length,
+				   url: '/api/usuarios'
+			    },
+            data: usuarioRespuesta
+        };
+        res.status(200).json(respuesta);
+    },
+	detalle: async (req, res) => {
+		let usuario = await usuarioRepository.ObtenerPorId(req.params.id);
+		if(usuario != null){
+            let respuesta = {
+                    meta: {
+                        url: '/api/usuarios/' + req.params.id
+                    },
+                    data: {
+						id: usuario.id,
+						nombre: usuario.nombre,
+						apellido: usuario.apellido,
+						email: usuario.email,
+						avatar: usuario.avatar,
+						roles: usuario.roles
+					}
+            };
+            res.status(200).json(respuesta);    
+	    }else{
+		    let respuesta = {
+			        meta: {
+				        url: '/api/usuarios/' + req.params.id
+			        },
+			        data: "id inexistente"
+			};
+            res.status(200).json(respuesta);    
```

Comment:
No estoy 100% seguro, esto podriamos preguntaselo a Pablo, pero aca el estus en lugar de 200 deberia ser 404, que es el codigo de "Not found"

Reply:
lo pruebo y te digo

Reply:
cris puse el 404 peor es lo mismo, ahi le pregunte a pablo a ver que me dice